LoopKit/Alert.swift

```diff
@@ -94,15 +94,21 @@ public struct Alert: Equatable {
         case sound(name: String)
     }
     public let sound: Sound?
+
+    /// Any metadata for the alert used to customize the alert content
+    public typealias MetadataValue = AnyCodableEquatable
+    public typealias Metadata = [String: MetadataValue]
+    public let metadata: Metadata?
     
     public init(identifier: Identifier, foregroundContent: Content?, backgroundContent: Content?, trigger: Trigger,
-                interruptionLevel: InterruptionLevel = .timeSensitive, sound: Sound? = nil) {
+                interruptionLevel: InterruptionLevel = .timeSensitive, sound: Sound? = nil, metadata: Metadata? = nil) {
         self.identifier = identifier
         self.foregroundContent = foregroundContent
         self.backgroundContent = backgroundContent
         self.trigger = trigger
         self.interruptionLevel = interruptionLevel
         self.sound = sound
+        self.metadata = metadata
     }
 }
 
```

Comment:
Just to make things a little more ergonomic, consider adding an extension so Metadata construction is a little cleaner:

```swift
extension Alert.Metadata {
    init<E: Codable & Equatable>(_ dict: [String: E]) {
        self = dict.mapValues { AnyCodableEquatable($0) }
    }
}
```
That way you can initialize `Metadata` like this:
```swift
let metadata = Alert.Metadata(["Foo": "Bar"])
```

Just a suggestion...

Reply:
This will work for simple cases (dictionary values are all of the same type), but not for more complex cases (mixed value type dictionary). However, added and used where possible.